From 8c16321b42876dde4dba4ff335032d92ce709eac Mon Sep 17 00:00:00 2001
From: Waltteri Hooli <wahooli@localhost>
Date: Sat, 15 Aug 2026 17:20:44 +0300
Subject: [PATCH 27/28] arm64: dts: qcom: hamoa: add system power domain SS3
 state

Port of Val Packett's unmerged patch 'arm64: dts: qcom: x1e80100: add
system power domain SS3 state' (lkml 2510.1/04745, arm64-laptops issue
16) onto 7.1's renamed hamoa.dtsi. Without domain_ss3 the system power
domain has no idle state, genpd never casts the RPMh system-sleep vote,
and aosd/cxsd/apss/ddr stay at 0 forever - s2idle drains at roughly
screen-off-idle power. Latency/residency numbers are copied from SM8750
(Konrad Dybcio notes they are phone-tuned; functional but possibly
suboptimal for a laptop).
---
 arch/arm64/boot/dts/qcom/hamoa.dtsi | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/hamoa.dtsi b/arch/arm64/boot/dts/qcom/hamoa.dtsi
index 4ba751a65..7b7f977ff 100644
--- a/arch/arm64/boot/dts/qcom/hamoa.dtsi
+++ b/arch/arm64/boot/dts/qcom/hamoa.dtsi
@@ -302,6 +302,14 @@ cluster_cl5: cluster-sleep-1 {
 				exit-latency-us = <4000>;
 				min-residency-us = <7000>;
 			};
+
+			domain_ss3: domain-sleep-0 {
+				compatible = "domain-idle-state";
+				arm,psci-suspend-param = <0x0200c354>;
+				entry-latency-us = <2800>;
+				exit-latency-us = <4400>;
+				min-residency-us = <10150>;
+			};
 		};
 	};
 
@@ -460,7 +468,7 @@ cluster_pd2: power-domain-cpu-cluster2 {
 
 		system_pd: power-domain-system {
 			#power-domain-cells = <0>;
-			/* TODO: system-wide idle states */
+			domain-idle-states = <&domain_ss3>;
 		};
 	};
 
